### ISSUE TITLE
Error when reading or editing Dataset: googleapi: Error 400: Dataset <resource-name> is still in use, resourceInUse

### ISSUE THREAD

**Issue by cdancy** [terraform-google-modules/terraform-google-log-export#135]
### TL;DR

We previously applied the terraform conditionally via a module (i.e. count = 1). Now when we run the same "setup code" and set the conditionally to 0 on said module we get a failure to delete the Dataset saying it's still in use. We're not exactly sure how to proceed here. Any help would be appreciated.

EDIT: deleting through the console works as expected but terraform is failing.

### Expected behavior

No errors to happen :)

### Observed behavior

We get the below error when trying to delete (forgive all the extra added markers from our automation code):
```
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mError when reading or editing Dataset: googleapi: Error 400: Dataset <resource-name> is still in use, resourceInUse[0m
[31m│[0m [0m
[31m│[0m [0m[0m
[31m╵[0m[0m
[31m╷[0m[0m


[31m│[0m [0m[1m[31mError: [0m[0m[1mError creating Dataset: googleapi: Error 409: Already Exists: Dataset <resource-name>, duplicate[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with module.integration.module.logging_framework[0].module.destination_bq.google_bigquery_dataset.dataset,
[31m│[0m [0m  on .terraform/modules/integration.logging_framework.destination_bq/modules/bigquery/main.tf line 41, in resource "google_bigquery_dataset" "dataset":
[31m│[0m [0m  41: resource "google_bigquery_dataset" "dataset" [4m{[0m[0m
```

### Terraform Configuration

```hcl
module "log_bq_sink" {
  source                 = "terraform-google-modules/log-export/google"
  destination_uri        = module.destination_bq.destination_uri
  log_sink_name          = local.bq_logsink_name
  filter                 = "resource.labels.project_id=${var.AccountID}"
  //exclusions = []
  bigquery_options       = {
                             use_partitioned_tables = true
                           }
  parent_resource_id     = var.AccountID
  parent_resource_type   = "project"
  unique_writer_identity = true  //Must be set to true for BQ option
}

module "destination_bq" {
  source                   = "terraform-google-modules/log-export/google//modules/bigquery"
  description              = "Data set which contains all logs generated from pega GKE cluster"
  project_id               = var.AccountID
  dataset_name             = local.bq_dataset_name
  location                 = var.Region
  log_sink_writer_identity = module.log_bq_sink.writer_identity
  expiration_days          = local.bq_expiration_in_days
  delete_contents_on_destroy = local.delete_contents_on_destroy
  labels = {
    project = var.AccountID
    owner = var.Owner
    location = var.Region
  }
}
```

### Terraform Version

```sh
1.0.11
```
